tests/object_file_system_test.php

```diff
@@ -563,35 +562,41 @@ public function test_presigned_url_should_redirect_method_return_true_regardless
         $this->filesystem = new test_file_system();
         $externalclient = $this->filesystem->get_external_client();
 
-        if ($externalclient->support_presigned_urls()) {
-            $this->set_externalclient_config('enablepresignedurls', '1');
-            $this->set_externalclient_config('presignedminfilesize', 0);
-            $object = $this->create_local_object();
-            $this->assertTrue($this->filesystem->presigned_url_should_redirect($object->contenthash));
+        if (!$externalclient->support_presigned_urls()) {
+            $this->markTestSkipped('Pre-signed URLs not supported for given storage.');
         }
+
+        $this->set_externalclient_config('enablepresignedurls', '1');
+        $this->set_externalclient_config('presignedminfilesize', 0);
+        $object = $this->create_local_object();
+        $this->assertTrue($this->filesystem->presigned_url_should_redirect($object->contenthash));
     }
 
     public function test_presigned_url_should_redirect_method_return_false_if_filesize_less_than_threshold() {
         $this->filesystem = new test_file_system();
         $externalclient = $this->filesystem->get_external_client();
 
-        if ($externalclient->support_presigned_urls()) {
-            $this->set_externalclient_config('enablepresignedurls', '1');
-            $this->set_externalclient_config('presignedminfilesize', 1000);
-            $object = $this->create_local_object();
-            $this->assertFalse($this->filesystem->presigned_url_should_redirect($object->contenthash));
+        if (!$externalclient->support_presigned_urls()) {
+            $this->markTestSkipped('Pre-signed URLs not supported for given storage.');
         }
+
+        $this->set_externalclient_config('enablepresignedurls', '1');
+        $this->set_externalclient_config('presignedminfilesize', 1000);
+        $object = $this->create_local_object();
+        $this->assertFalse($this->filesystem->presigned_url_should_redirect($object->contenthash));
```

Comment:
why is it false here? does create_local_object create the file less than 1000? If it is, can you please add some comment here?

Reply:
It is false because local test files are created with file size = 10:
https://github.com/catalyst/moodle-tool_objectfs/blob/issue176/tests/tool_objectfs_testcase.php#L86